pyo3-ffi/src/pythread.rs

```diff
@@ -0,0 +1,98 @@
+use crate::object::PyObject;
+use std::os::raw::{c_int, c_longlong, c_ulong, c_void};
+
+pub type PyThread_type_lock = *mut c_void;
+// skipped PyThread_type_sema (removed 3.9)
+
+#[cfg(not(PyPy))]
+#[repr(C)]
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum PyLockStatus {
+    PY_LOCK_FAILURE = 0,
+    PY_LOCK_ACQUIRED = 1,
+    PY_LOCK_INTR,
+}
+
+extern "C" {
+    pub fn PyThread_init_thread();
+    pub fn PyThread_start_new_thread(
+        arg1: Option<unsafe extern "C" fn(*mut c_void)>,
+        arg2: *mut c_void,
+    ) -> c_ulong;
+
+    #[cfg(not(PyPy))]
+    pub fn PyThread_exit_thread() -> !;
```

Comment:
`PyThread_exit_thread` is deprecated in 3.14 and the docs say it shouldn't be used, let's skip this one.